soc/xtensa/intel_adsp/common/multiprocessing.c

```diff
@@ -96,6 +96,24 @@ __asm__(".section .text.z_soc_mp_asm_entry, \"x\" \n\t"
 #undef NOP32
 #undef NOP4
 
+__asm__(".align 4\n\t"
+	".global dsp_restore_vector\n\t"
+	"dsp_restore_vector:\n\t"
```

Comment:
Design note:

This is like the fifth or possibly sixth mostly-cut-and-paste copy of this bootstrap code.  It's an inline inside C code only because the first version (which I did for cavs15, I think?) was ~5 instructions and didn't need to be anything fancy.  But at this point it's grown well beyond that, and every version has its own hackery (like the IMHO senseless "compute a runtime array index based on processor ID to get our stack pointer").

What this really wants to be is some kind of build-time gadget (maybe a C macro, maybe a gas/asm macro, maybe a python script, maybe even a set of .S files generated from cmake) that does something like "*Please emit a bootstrap sequence that can be jumped to via this symbol name, that sets PS to this value, that jumps to this symbol after setup, and that uses this stack pointer*"